google-domain-user-picker.html

```diff
@@ -0,0 +1,459 @@
+    .result-container:hover {
+      background-color: #FAFAFA;
+    }
+
+    [focused] {
+      background-color: #EEEEEE;
+    }
+
+    .result-user-container {
+      margin-left: 10%;
+      @apply(--layout-flex);
+    }
+
+  </style>
+
+  <template>
+
+    <google-signin-aware
+        scopes="https://www.googleapis.com/auth/admin.directory.user.readonly"
+        on-google-signin-aware-success="_onSignIn">
+    </google-signin-aware>
+
+    <iron-ajax id="ajax"
+               url="https://www.googleapis.com/admin/directory/v1/users"
+               method="GET"
+               params="{{_getRequestParams(value, accessToken)}}"
+               headers="{{_getRequestHeaders(accessToken)}}"
+               handle-as="json"
+               debounce-duration="50"
+               on-response="_handleResponse">
+    </iron-ajax>
+
+    <iron-localstorage name="google-domain-user-picker-frequently-picked"
+                       value="{{frequentlyPicked}}">
+    </iron-localstorage>
+
+    <div class="search-container">
+      <iron-icon icon="search"></iron-icon>
+      <input is="iron-input"
+             label="{{placeholder}}"
+             bind-value="{{value}}"
+             on-focus="_onFocus"
+             placeholder="{{placeholder}}"
+             class="search">
+      </iron-input>
+    </div>
+
+    <div id="results" role="menu" tab-index="0">
+      <template id="resultsTemplate" is="dom-repeat" items="{{people}}">
+        <div class="result-container"
+             on-tap="_handleUserSelect"
+             focused$="{{_isFocused(item, _focusedIndex)}}">
+          <img src="{{item.image}}">
+          <div class="result-user-container">
+            <div>{{item.name}}</div>
+          </div>
+        </div>
+      </template>
+    </div>
+
+  </template>
+
+</dom-module>
+
+<script>
+(function() {
+  Polymer({
+
+    is: 'google-domain-user-picker',
+
+    /**
+     * fired when a user is selected by the mouse or enter key
+     *
+     * @event picker-select-user
+     * @detail {{name: String, email: String, id: String, image: String}}
+     */
+
+    /**
+     * fired when a user is focused by the mouse or arrow keys
+     *
+     * @event picker-focus-user
+     * @detail {{name: String, email: String, id: String, image: String}}
+     */
+
+    /**
+     * fired when the user presses the escape key
+     * developers should use this to hide the element
+     *
+     * @event picker-esc
+     */
+
+    /**
+     * fired when there is an error with the picker
+     * for example, when there is no network connectivity
+     *
+     * @event picker-error
+     * @detail {String} the human-readable reason for the error
+     */
+
+    properties: {
+
+      /**
+       * `accessToken` is the accessToken that will be used when making
+       * authenticated requests to the Google Admin SDK API
+       *
+       * @required
+       * @property accessToken
+       * @type String
+       */
+      accessToken: {
+        type: String,
```

Comment:
Feels a bit weird you're exposing `accessToken` as a property, but I guess that's because you want it usable with just the core js lib?